### PR TITLE
Fix ASTPHP5Scanner (Heredoc & Nowdoc syntaxes)

### PR DESCRIPTION
**Pull request by junichi11**
- Fix unit test results for Heredoc and Nowdoc because they are wrong
- So, fix ASTPHP5Scanner, too

### CODE REVIEW

File: php/php.editor/test/unit/data/goldenfiles/org/netbeans/modules/php/editor/parser/ASTPHP5ParserTest/parser/heredoc00.pass [hunk 1]
```diff
@@ -42,11 +42,11 @@
...
         </token>
-        <token id='T_ENCAPSED_AND_WHITESPACE' start='73' end='84'>
-            <text>&lt;br&gt;\nHEREDO</text>
+        <token id='T_ENCAPSED_AND_WHITESPACE' start='73' end='78'>
+            <text>&lt;br&gt;\n</text>
         </token>
-        <token id='T_END_HEREDOC' start='84' end='85'>
-            <text>C</text>
+        <token id='T_END_HEREDOC' start='78' end='85'>
+            <text>HEREDOC</text>
```
Comment (junichi11) [inline php/php.editor/test/unit/data/goldenfiles/org/netbeans/modules/php/editor/parser/ASTPHP5ParserTest/parser/heredoc00.pass]: Token texts were not correct like this.
https://github.com/apache/incubator-netbeans/blob/4fe72ac68f1b22311365b3e91315d4c267c9d638/php/php.editor/test/unit/data/testfiles/parser/heredoc00.php#L1-L24

File: php/php.editor/tools/ASTPHP5Scanner.flex [hunk 2]
```diff
@@ -72,8 +72,14 @@ import org.netbeans.modules.web.common.api.ByteStack;
 %{
     private final List commentList = new ArrayList();
     private String heredoc = null;
+    private int heredocBodyStart = -1;
+    private int heredocBodyLength = 0;
+    private final StringBuilder heredocBody = new StringBuilder();
```
Comment (junichi11) [inline php/php.editor/tools/ASTPHP5Scanner.flex]: For getting a T_ENCAPSED_AND_WHITESPACE(heredoc string) as one token.

File: php/php.editor/tools/ASTPHP5Scanner.flex [hunk 3]
```diff
@@ -72,8 +72,14 @@ import org.netbeans.modules.web.common.api.ByteStack;
...
     private String nowdoc = null;
     private int nowdoc_len  = 0;
+    private int nowdocBodyStart = -1;
+    private int nowdocBodyLength = 0;
+    private final StringBuilder nowdocBody = new StringBuilder();
```
Comment (junichi11) [inline php/php.editor/tools/ASTPHP5Scanner.flex]: For getting a T_ENCAPSED_AND_WHITESPACE(nowdoc string) as one token.

**Comment by geertjanw** [comment]
Great, thanks. Should this be merged, should we wait?

**Comment by tmysik** [comment]
Merged.

**Pull request merged**